lisp/magit-branch.el

```
@@ -201,6 +201,9 @@ has to be used to view and change branch related variables."
 (transient-define-prefix magit-branch (branch)
   "Add, configure or remove a branch."
   :man-page "git-branch"
+  :value '("--recurse-submodules")
+  ["Arguments"
+   ("-r" "Recurse submodules when checking out an existing branch" "--recurse-submodules")]
```

This shouldn't be enabled by default because it detaches HEAD and I think it should even be hidden by default.

---

ok, I can see that detaching HEAD might break other people's workflows. I'll check how I can make default-enabling and default-showing this configurable via the magit-group options, as I'd very much like to have that for my workflow.

btw, what is the rationale for default-hiding? just uncluttering the interface for a probably rare use-case or something eles in addition to that?
It would probably also make sense to list the option to default-show this option underneath the magit-submodule documentation in addition to the magit-branch documentation, wouldn't it? at least that is where I looked first for something like this.

---

Additional things to note when considering whether to show this by default:

  * the outstanding bug that I linked to in my other reply

  * this isn't available until 2.13 (which is more recent than magit's current minimum git version)

    Perhaps it's worth guarding this option in the same way the bisect transient does:
    `:if (lambda () (version<= "2.13" (magit-git-version)))`

---

thanks for the feedback, I'll try to guard the option and not enable/show this by default as soon as I've fixed some of the other findings here

---

I've added user options to show and default-enable this, in addition the git-version is checked as proposed. If that is ok, I'd take a shot at updating the user documentation so that people are aware of these options.

---

I believe this is solved now. I will resolve the GitHub conversation if no one complains soon